src/utils/spline_builder_deriv_field_2d.hpp

```diff
@@ -0,0 +1,305 @@
+                });
+    }
+
+    /**
+     *  @brief Fill in the deriv2 field with the derivatives along the second dimension stored 
+     * in the function_and_derivs. 
+     * @param[out] deriv2 Field with layout_right where we copy the derivatives.
+     * @param[in] function_and_derivs DerivField from where the derivatives are copied.
+     * @param[in] is_min Boolean which helps to determine which bound (mon/max) we select for the derivative field. 
+     */
+    void fill_in_deriv2(Deriv2Field deriv2, DerivFieldType function_and_derivs, bool const is_min)
+            const
+    {
+        IdxRangeSlice<Grid2> idx_range_slice_2
+                = function_and_derivs.template idx_range_for_deriv<Grid2>();
+        Idx<Grid2> idx_slice = is_min ? idx_range_slice_2.front() : idx_range_slice_2.back();
+
+        // Fill the field with correct layout.
+        ddc::parallel_for_each(
+                ExecSpace(),
+                get_idx_range(deriv2),
+                KOKKOS_LAMBDA(Idx<Grid1, Deriv2> const idx) {
+                    Idx<Grid1> idx_1(idx);
+                    Idx<Deriv2> idx_d2(idx);
+                    deriv2(idx) = function_and_derivs(idx_d2, idx_slice, idx_1);
+                });
+    }
+
+    /**
+     *  @brief Fill in the cross_deriv field with the cross derivatives stored 
+     * in the function_and_derivs. 
+     * @param[out] cross_deriv Field with layout_right where we copy the cross-derivatives.
+     * @param[in] function_and_derivs DerivField from where the cross-derivatives are copied.
+     * @param[in] is_1min Boolean which helps to determine which bound (mon/max) we select for the derivative field
+     * on the first dimension. 
+     * @param[in] is_2min Boolean which helps to determine which bound (mon/max) we select for the derivative field
+     * on the second dimension. 
+     */
+    void fill_in_cross_deriv(
+            CrossDerivField cross_deriv,
+            DerivFieldType function_and_derivs,
+            bool const is_1min,
+            bool const is_2min) const
+    {
+        static_assert(Builder2D::builder_type1::s_nbc_xmin == Builder2D::builder_type1::s_nbc_xmax);
+        static_assert(Builder2D::builder_type2::s_nbc_xmin == Builder2D::builder_type2::s_nbc_xmax);
+
+        IdxRangeSlice<Grid1> idx_range_slice_1
+                = function_and_derivs.template idx_range_for_deriv<Grid1>();
+        IdxRangeSlice<Grid2> idx_range_slice_2
+                = function_and_derivs.template idx_range_for_deriv<Grid2>();
+
+        IdxRange<Deriv1, Deriv2> idx_range_derivs(function_and_derivs.derivative_idx_range());
+        Idx<Grid1> idx_slice_1 = is_1min ? idx_range_slice_1.front() : idx_range_slice_1.back();
+        Idx<Grid2> idx_slice_2 = is_2min ? idx_range_slice_2.front() : idx_range_slice_2.back();
+
+        // Fill the field with correct layout.
+        ddc::parallel_for_each(
+                idx_range_derivs,
```

Comment:
I think both are valid but why did you decide to use `idx_range_derivs` here and `get_idx_range(deriv2)` in the function above? I guess you could use `get_idx_range` here too right?